examples/go-sdk-workflow/main.go

```diff
@@ -57,7 +62,38 @@ func main() {
 		panic(err)
 	}
 
-	err = w.On(
+	w.Use(func(ctx context.Context, next func(context.Context) error) error {
+		err := next(context.WithValue(ctx, "testkey", "testvalue"))
+
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+
+	w.Use(func(ctx context.Context, next func(context.Context) error) error {
+		// time the function duration
+		start := time.Now()
+		err := next(ctx)
+		duration := time.Since(start)
+		fmt.Printf("step function took %s\n", duration)
+		return err
+	})
+
+	testSvc := w.NewService("test")
+
+	testSvc.Use(func(ctx context.Context, next func(context.Context) error) error {
+		err := next(context.WithValue(ctx, "svckey", "svcvalue"))
+
+		if err != nil {
+			return err
+		}
+
+		return nil
```

Comment:
Could be shortened to `return next(...)`, but maybe it's better to keep it verbose as it is an example